redash/serializers.py

```diff
@@ -9,6 +9,7 @@
 from flask_login import current_user
 from redash import models
 from redash.permissions import has_access, view_only
+from redash.handlers.query_results import run_query_sync
```

Comment:
> [Please don't reuse in other API handlers.](https://github.com/getredash/redash/blob/master/redash/handlers/query_results.py#L22-L25)

:-)

There are two issues with `run_query_sync`:

1. Security issue. It allows SQL injections for external users, which you avoid by not supporting parameters.
2. Performance: it runs synchronously, so if there are enough requests it can block the web server.

You avoided the security issue by not supporting parameters, but the performance (and in a way another security issue) is still there.